packages/thirdweb/src/transaction/prepare-transaction.ts

```diff
@@ -74,6 +74,11 @@ export type PreparedTransaction<
 > = Readonly<options> & {
   __preparedMethod?: () => Promise<PreparedMethod<abiFn>>;
   __contract?: ThirdwebContract<abi>;
+  // additional
+  valueERC20?: {
+    amount: string; // units of tokens
+    tokenAddress: string;
+  };
```

Comment:
shouldn't this be in `PrepareTransactionOptions` ?

Reply:
Added on `PreparedTransaction` to allow mutating them to set them after getting tx object from extension

Reply:
this pattern doesnt feel right. 

we should just add it as optional inputs to prepareTransaction / prepareContractCall and the overrides for extensions. Same way as you pass native value currently.